src/sync/sync_permission.cpp

```diff
@@ -59,6 +60,16 @@ Permission::AccessLevel extract_access_level(Object& permission, CppContext& con
     return Permission::AccessLevel::None;
 }
 
+/// Turn a system time point value into the 64-bit integer representing ns since the Unix epoch.
+int64_t ns_since_unix_epoch(const system_clock::time_point& point)
+{
+    tm unix_epoch = {0};
+    unix_epoch.tm_year = 70;
+    time_t epoch_time = mktime(&unix_epoch);
+    auto epoch_duration = system_clock::from_time_t(epoch_time).time_since_epoch();
+    return duration_cast<nanoseconds>(point.time_since_epoch() + epoch_duration).count();
```

Comment:
Shouldn't you be subtracting the UNIX epoch from the current time rather than adding it?

I don't think it's necessary to call `time_since_epoch()` either, since there's an `operator-` defined for two `std::time_point`s that gives you the duration between them.